controllers/storagecluster/storageclient.go

```
@@ -27,11 +24,17 @@ func (s *storageClient) ensureCreated(r *StorageClusterReconciler, storagecluste
 		return s.ensureDeleted(r, storagecluster)
 	}
 
+	privateKey, err := util.ReadPrivateKey(r.Client)
+	if err != nil {
+		r.Log.Error(err, "Unable to get privatekey")
+		return reconcile.Result{}, nil
```

you might want to return an error here.

---

ack